applications/FluidDynamicsApplication/custom_elements/d_vms_dem_coupled.h

```diff
@@ -0,0 +1,394 @@
+    typedef Kratos::Matrix ShapeFunctionDerivativesType;
+
+    /// Type for an array of shape function gradient matrices
+    typedef GeometryType::ShapeFunctionsGradientsType ShapeFunctionDerivativesArrayType;
+
+    constexpr static unsigned int Dim = DVMS<TElementData>::Dim;
+    constexpr static unsigned int NumNodes = DVMS<TElementData>::NumNodes;
+    constexpr static unsigned int BlockSize = DVMS<TElementData>::BlockSize;
+    constexpr static unsigned int LocalSize = DVMS<TElementData>::LocalSize;
+    constexpr static unsigned int StrainSize = DVMS<TElementData>::StrainSize;
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    //Constructors.
+
+    /// Default constuctor.
+    /**
+     * @param NewId Index number of the new element (optional)
+     */
+    DVMSDEMCoupled(IndexType NewId = 0);
+
+    /// Constructor using an array of nodes.
+    /**
+     * @param NewId Index of the new element
+     * @param ThisNodes An array containing the nodes of the new element
+     */
+    DVMSDEMCoupled(IndexType NewId, const NodesArrayType& ThisNodes);
+
+    /// Constructor using a geometry object.
+    /**
+     * @param NewId Index of the new element
+     * @param pGeometry Pointer to a geometry object
+     */
+    DVMSDEMCoupled(IndexType NewId, GeometryType::Pointer pGeometry);
+
+    /// Constuctor using geometry and properties.
+    /**
+     * @param NewId Index of the new element
+     * @param pGeometry Pointer to a geometry object
+     * @param pProperties Pointer to the element's properties
+     */
+    DVMSDEMCoupled(IndexType NewId, GeometryType::Pointer pGeometry, Properties::Pointer pProperties);
+
+    /// Destructor.
+    ~DVMSDEMCoupled() override;
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+
+    /// Create a new element of this type
+    /**
+     * Returns a pointer to a new DVMSDEMCoupled element, created using given input
+     * @param NewId the ID of the new element
+     * @param ThisNodes the nodes of the new element
+     * @param pProperties the properties assigned to the new element
+     * @return a Pointer to the new element
+     */
+    Element::Pointer Create(IndexType NewId,
+                            NodesArrayType const& ThisNodes,
+                            Properties::Pointer pProperties) const override;
+
+    /// Create a new element of this type using given geometry
+    /**
+     * Returns a pointer to a new DVMSDEMCoupled element, created using given input
+     * @param NewId the ID of the new element
+     * @param pGeom a pointer to the geomerty to be used to create the element
+     * @param pProperties the properties assigned to the new element
+     * @return a Pointer to the new element
+     */
+    Element::Pointer Create(IndexType NewId,
+                            GeometryType::Pointer pGeom,
+                            Properties::Pointer pProperties) const override;
+
+    void Calculate(
```

Comment:
Did you really need to override all these 'Calculate' functions?

Reply:
Agree. Indeed in your implementation you are precluding the access to the base class ones.